note-service/src/main/java/ma/enset/noteservice/controller/NoteElementController.java

```diff
@@ -92,6 +94,27 @@ public ResponseEntity<NoteElementResponse> update(
                 .body(updatedModuleResponse);
     }
 
+    @PatchMapping("/bulk")
+    public ResponseEntity<List<NoteElementResponse>> updateMany(
+            @NotEmpty @RequestParam("noteElementId") List<String> noteElementIds,
```

Comment:
as said before, you can include ids inside updateRequest body, and the RequestBody must be List<NoteElementUpdateRequest>, we are updating multiple objects, no one.

Reply:
I think it's better to go with request param, and I'm updating a list of object, not one!

Reply:
you are passing list of ids and a single object. you update all the objects associated with those ids with the updateRequest you send in body. that isn't right. you must provide list of updateRequest and for each id you update its object with one of the updateRequests . (checkout bulk update in other services)